npc/quests/skills/blacksmith_skills.txt

```diff
@@ -15,7 +15,7 @@
 
 geffen,178,72,3	script	Akkie#qsk_bs	726,{
 	if (BaseJob == Job_Blacksmith) {
-		if (BLACK_SK == 7 && getskilllv("BS_UNFAIRLYTRICK") == 0) {
+		if (BLACK_SK >= 7 && Upper == 1 && getskilllv("BS_UNFAIRLYTRICK") == 0) {
```

Comment:
On aegis
- `BLACK_SK` 7 and 8, `brack_greed` 2 and 3 can only be reachable by rebirth and transcendant 3rd class
- neuralizer doesn't reset quests skills

=
- add a condition for theses variable to only be reachable by rebirth and transcendant 3rd class
- I would prefer to simply check BLACK_SK case 7 for non rebirth and case 8 for transcendant if the skill is learned and give it accordly